### ISSUE TITLE
Downloading from private repos

### ISSUE THREAD

**Issue by mbevc1** [marcosnils/bin#121]
Using `GITHUB_AUTH_TOKEN` only resolves metadata, but doesn't fully work/download when using private repos. It's trying to download from `https://github.com/org/repo/releases/download/version/binary` and that would require pulling it down from assets URL rather than `browser_download_url`.

Example using `curl`:

```
curl -H "Authorization: token ghp_xxxxxxxxxx" -H "Accept: application/octet-stream" -O -L https://api.github.com/repos/org/repo/releases/assets/1111111
```

**Comment by marcosnils** [comment]
:wave: just released 0.11.0 with this functionality! thx for reporting this!

**Comment by mbevc1** [comment]
Thanks for quick response @marcosnils! I'll give it a go :+1:
